utils.py

```diff
@@ -76,13 +138,40 @@ def dataset_from_folder(folder, img_ext='.png', mask_ext='.trn'):
         yield load_image_for_dataset(file)
 
 if __name__ == '__main__':
-    for datum in dataset_from_folder('./plzen'):
+    data, data_mask = load_dataset('./plzen')
```

Comment:
@Foxiy I really do not like the fact that in the actual usecase we would read the whole dataset twice, just to run augmentation in batches. 

The big win of `dataset_from_folder` function's use of `yield` is in that if things get to be processed sequentially, this is extra fast and effective.

I agree though that for our usecase (actually trying to learn some models from the augmented data) it would be best to have the whole thing in memory and just doing batch operations on it.

Do you think we could have a function called `load_datase_from_folder` that would do just that (I mean load the data from disk to memory and do augmentation)? If someone did not want augmentation, they could just pass `augmentation=False` as a keyword parameter.

Ditching the `dataset_from_folder` function is totally on the table too. I do not want to invest time into something that will not be used after all, but I really hate when things are being done twice for no apparent reason.